pkg/ksync/cli.go

```diff
@@ -11,39 +11,58 @@ import (
 	"github.com/jenkins-x/jx/pkg/util"
 )
 
-// InstallKSync install ksync
-func InstallKSync() (string, error) {
+// KSyncer contains the methods used for ksync
+type KSyncer interface {
+	GetJXBinLocation() (string, error)
+	IsInstalled(string) (string, bool, error)
+	RunCommand(name string, args []string) (string, error)
```

Comment:
Is the intention do execute _ksync_ commands here? Or any type of command. If the former, I'd expect that 'ksync' (aka the main executable) is implicit in a _KSyncer_ interface. If the aim is to execute any arbitrary command then why does it have to be in the interface?

Reply:
Refactored the interface, it only has ksync commands (version, init and clean) atm. I did not add the other commands to the interface because I feel they are beyond the scope of this PR, I think it makes sense to make an issue which addresses cleaning up and refactoring the ksync codebase, and make the changes as part of that PR. what do u think?